src/methods/convex_hull.jl

```diff
@@ -0,0 +1,92 @@
+#=
+# Convex hull
+
+The _convex hull_ of a set of points is the smallest **convex** polygon that contains all the points.
+
+GeometryOps.jl provides a number of methods for computing the convex hull of a set of points, usually
+linked to other Julia packages.  
+
+## Example
+
+### Simple hull
+```@example simple
+import GeometryOps as GO, GeoInterface as GI
+using CairoMakie # to plot
+
+points = randn(GO.Point2f, 100)
+f, a, p = plot(points; label = "Points")
+hull_poly = GO.convex_hull(points)
+lines!(a, hull_poly; label = "Convex hull", color = Makie.wong_colors()[2])
+axislegend(a)
+f
+```
+
+## Convex hull of the USA
+```@example usa
+import GeometryOps as GO, GeoInterface as GI
+using CairoMakie # to plot
+using NaturalEarth # for data
+
+all_adm0 = naturalearth("admin_0_countries", 110)
+usa = all_adm0.geometry[findfirst(==("USA"), all_adm0.ADM0_A3)]
+f, a, p = lines(usa)
+lines!(a, GO.convex_hull(usa); color = Makie.wong_colors()[2])
+f
+```
+
+=#
+
+"""
+    convex_hull([method], geometries)
+
+Compute the convex hull of the points in `geometries`.  
+Returns a `GI.Polygon` representing the convex hull.
+
+Note that all 
+
+!!! warning
+    This interface only computes the 2-dimensional convex hull!
+
+    For higher dimensional hulls, use the relevant package (Qhull.jl or similar).
+"""
+function convex_hull end
+
+"""
+    DelaunayTriangulationMethod()
+
+This is an algorithm for the [`convex_hull`](@ref) function.
+
+Uses [`Delaunator.jl`](https://github.com/JuliaGeometry/DelaunayTriangulation.jl) to compute the convex hull.
+This is a pure Julia algorithm which provides an optimal Delaunay triangulation.
+
+See also [`convex_hull`](@ref)
+"""
+struct DelaunayTriangulationMethod end
+
+# , [`DelaunayTriangulationMethod`](@ref), [`EarCutMethod`](@ref).
+# GrahamScanMethod, etc. can be implemented in GO as well, if someone wants to...
+
+convex_hull(geometries) = convex_hull(DelaunayTriangulationMethod(), geometries)
+
+# TODO: have this respect the CRS by pulling it out of `geometries`.
+function convex_hull(::DelaunayTriangulationMethod, geometries)
+    # Extract all points as tuples.  We have to collect and allocate
```

Comment:
Is this true? You can provide whatever point types you like so long as you extend the interfaces (which are hopefully sufficiently documented) from DelaunayTriangulation.jl appropriately. I give a complete example here https://juliageometry.github.io/DelaunayTriangulation.jl/dev/tutorials/custom_primitive/, maybe something goes wrong in `convex_hull` that I'm not aware of?

Reply:
We have to flatten anyway to get a vector, and getting x and y values from C pointers is pretty bad for performance - best to do it once.  I can update the comment with better verbiage, though.

Reply:
Plus, we'd have to extend the interface for every GeoInterface compatible type, which would probably mean either piracy or changes on the DelaunayTriangulation end.  (Though if you do want to allow GeoInterface as a fallback it would be pretty easy I think).

Reply:
> We have to flatten anyway to get a vector, and getting x and y values from C pointers is pretty bad for performance - best to do it once. I can update the comment with better verbiage, though.

I have no preference either way just wanted to see if you were aware (I wasn't aware of the need for getting the values from C pointers, so fair enough). The verbiage is alright

> Plus, we'd have to extend the interface for every GeoInterface compatible type, which would probably mean either piracy or changes on the DelaunayTriangulation end. (Though if you do want to allow GeoInterface as a fallback it would be pretty easy I think).

It would probably be welcome as a package extension in DelaunayTriangulation. I'm not familiar with the interfaces though, so I probably won't be delving into that. Just leaving the code as is is fine anyway, it works and would be fast enough :).

Reply:
It would probably just be:
```julia
DT.getx(p) = DT.getx(GI.trait(p), p)
DT.gety(p) = DT.gety(GI.trait(p), p)
DT.number_type(p) = DT.number_type(GI.trait(p), p)

DT.getx(::GI.PointTrait, p) = GI.x(p)
DT.gety(::GI.PointTrait, p) = GI.y(p)
DT.number_type(::GI.PointTrait, p) = typeof(GI.x(p))